### ISSUE TITLE
libreoffice build is broken (test failure)

### ISSUE THREAD

**Issue by bjornfor** [NixOS/nixpkgs#16991]
## Issue description
libreoffice used to build, now it doesn't. IMHO, this is a blocking issue for desktop use.

### Steps to reproduce
```
$ nix-build -A libreoffice
...
[build CUT] sc_new_cond_format_api
[build CUT] sc_subsequent_filters_test
[build CUT] sc_subsequent_export_test
[build CUT] sc_bugfix_test
W: Unknown node under /registry/extlang: deprecated
W: Unknown node under /registry/grandfathered: comments
W: Unknown node under /registry/grandfathered: comments
csv Test
xlsx Test
/tmp/nix-build-libreoffice-5.0.6.3.drv-0/libreoffice-5.0.6.3/sc/qa/unit/subsequent_filters-test.cxx:1601:ScFiltersTest::testChartImportXLS
equality assertion failed
- Expected: 11148
- Actual  : 12045

subsequent_filters-test.cxx:1601:Assertion
Test name: ScFiltersTest::testChartImportXLS
equality assertion failed
- Expected: 11148
- Actual  : 12045

Failures !!!
Run: 79   Failure total: 1   Failures: 1   Errors: 0
which: no gdb in (/nix/store/idgb6m80hgxgj8ay9k005pjdv411k7pc-ant-1.9.6/bin:/nix/store/hyf4p55xhz2glljkmizh0s13rvpnv496-perl-Archive-Zip-1.16/bin:/nix/store/4m012rlrihgn26d3wilq2bz87417f09n-autoconf-2.69/bin:/nix/store/rzv37i7lmnahb717gxi0xk1hribvmnd7-automake-1.15/bin:/nix/store/gyjfjwznqq36x3yjpbmlc762y88njgz4-bison-3.0.4/bin:/nix/store/5fk506wsiw3hwyjw53il5y73s834391s-gnum4-1.4.17/bin:/nix/store/bimqgjghsdisi3ygxdynlhkd7xsrnhh2-cairo-1.14.6-dev/bin:/nix/store/7qgrdx3xczhahdwsv5rfq9xf4yyh71rw-freetype-2.6.2-dev/bin:/nix/store/gcdzwrifzi2y9dg7mrg9ra4nm2phnkiz-bzip2-1.0.6.0.1-bin/bin:/nix/store/x18dnp1pvn2279isj0cv1cq75m2m0znc-libpng-apng-1.6.23-dev/bin:/nix/store/69cajn9kqk4w966sdi9hg83d92pzp9kh-fontconfig-2.11.1-bin/bin:/nix/store/8nlrlf5cb816nkkk4ybi2gpdlx96sp80-expat-2.2.0-dev/bin:/nix/store/4i3742qdisav4ri7iy4m44vl9sw4qbsa-glib-2.48.1-dev/bin:/nix/store/snxbisn3r0cb9k3xbdc73ady3msxb9wc-glibc-2.23-bin/bin:/nix/store/3b1yl1ar23miy1a33iqnfrs9yfh6f093-perl-IO-Compress-2.063/bin:/nix/store/5jqbj3csx6py4pllx0lkrwf07bgf3n5a-cppunit-1.13.2/bin:/nix/store/bvfjzacdbhq9rafl5i0333kxwsb48f92-cups-2.1.3-dev/bin:/nix/store/pq40s2hq37l7wyzcxgpkb4fj1nhp3zf5-cups-2.1.3/bin:/nix/store/axbpd4x2x2i7c0pkxhxvikv5n3dfbx8a-curl-7.47.1-dev/bin:/nix/store/kgd8sbl0qwamj1iqaia3f8ra1jb0g0d3-nghttp2-1.10.0/bin:/nix/store/1p0bxk9sm1b7h00vag9h45nk77zkf9z8-openssl-1.0.2h-bin/bin:/nix/store/mpqacjnsa46hb4gpr1fzb5q4k61vn09y-curl-7.47.1-bin/bin:/nix/store/679v42s69gx5083irha8r1162l2iw5vc-db-5.3.28/bin:/nix/store/l4bfj3hz18qkc170b633i9z45j4iapqx-dbus-glib-0.106-dev/bin:/nix/store/j1i3c1n8j9i3npzgm21wx616a6jh0h0d-dbus-1.10.8-lib/bin:/nix/store/mkmfmxgscz2r6kj8d05k2rswrxn22fmw-dbus-1.10.8/bin:/nix/store/sc0ckhfdidank1vzfx5996kgch02p32v-file-5.28/bin:/nix/store/pzg26fydpfn0l6fx03fapwqbai3v44c3-flex-2.6.0/bin:/nix/store/yyd92q5ia737pf74c5r2h66dv8n92fbl-gconf-2.32.4/bin:/nix/store/ilcwn940pj4qzwis8gpfrh6i0a52bsfk-getopt-1.1.4/bin:/nix/store/cqml6ykw5kaq5yc7nfp0363bzjxcha6y-gnome-vfs-2.24.4/bin:/nix/store/vz7p25jwfrnqy34haic7y0a1lafkpmpf-gperf-3.0.4/bin:/nix/store/ypandf5bn1gqfwl7cmxlzal65y6h5f4j-gtk+3-3.20.6-dev/bin:/nix/store/jw9fwa0wspisf19jslnma0vkfmpk1yg1-harfbuzz-1.2.7-dev/bin:/nix/store/qsnmqybp8hjqfzw6k6z31p6d3sshq6vi-graphite2-1.3.6/bin:/nix/store/5x7f85ik83g21xvqk9bxvzx5gbdrc90c-pango-1.40.1-bin/bin:/nix/store/yg821mhppk666fzcl31q5ik69rp0hpsm-gdk-pixbuf-2.34.0-dev/bin:/nix/store/5ck4ral81srkrhbj0v1irpa440hldilx-libjpeg-turbo-1.5.0-bin/bin:/nix/store/ca12bylrlmkdlzhi0k729xkrv68qx306-xz-5.2.2-bin/bin:/nix/store/wn1llpq7p4bdnmsrpjpfm797c8qz1p5a-libtiff-4.0.6-bin/bin:/nix/store/7l2zq8l8zamsh7icm00nv2gqgrx2bbdh-jasper-1.900.1-bin/bin:/nix/store/zvz0s1848i2kcyvm32yhcg6qbdl7mq51-gobject-introspection-1.48.0-dev/bin:/nix/store/d580inf10202x94d20l2mr1789zvp6jk-wayland-1.11.0/bin:/nix/store/a9afz3cvnflj6d59csb6ldrm30q519qa-gtk+-2.24.30-dev/bin:/nix/store/14mca3gbdylj5qp2i3svdjhyr8hf13if-hunspell-1.3.3-bin/bin:/nix/store/zpnclwc22819hl2j006fwjf4q3pyfz9f-icu4c-56.1-dev/bin:/nix/store/jgn2whwk2a92y2wysb64kyrwfr5c9xkg-openjdk-8u102b04/bin:/nix/store/gaqrwlkrng4kmw97a43w13icrws854fk-openjdk-8u102b04-jre/bin:/nix/store/smxnx5mvsnpq5i6v32l83dii9cyj74sg-lcms2-2.7-bin/bin:/nix/store/x109bxm6494n17gi2g0hlvb2f2x23qc6-libcdr-0.1.1/bin:/nix/store/31r61bx2sgrhladj063rnkwzbswn2di1-libexttextcat-3.4.1/bin:/nix/store/psbv12mdkdqlb56hks633hrqwm4p5bq8-unixODBC-2.3.4/bin:/nix/store/ggqps542cxw7p4lgb2ip5xggbjwf8226-redland-1.0.17/bin:/nix/store/xyk0vajdli2hckzhvic2gxx9lcnz9mbx-rasqal-0.9.32/bin:/nix/store/06vk1k3s644ibsv1n8qxwb2y4brk3v4z-raptor2-2.0.15/bin:/nix/store/pz0f4r4ihbaii3i3ddmm32vpi1765040-librsvg-2.40.9/bin:/nix/store/rfgwgf8cwm7nlx2xr2amf8nsd29bsf1n-libsndfile-1.0.26-bin/bin:/nix/store/n45205rppkllxwsrap9ccis5zgwgi0c1-libvisio-0.1.3/bin:/nix/store/63lm17263n1cplx7i16nh4xclviryv47-libwpd-0.10.0/bin:/nix/store/bcbl7a2zsgvgladvayh4x89ksb50ghb5-libwpg-0.3.0/bin:/nix/store/k8lgl2gm45577wdar5p8lfjh686xyf5s-libxml2-2.9.4-dev/bin:/nix/store/w1zydlzarx1s00sd8kdcbapzzdkxjr6a-libxml2-2.9.4-bin/bin:/nix/store/wxgd98bsx116z7djw18m2kpdbbx8z9ns-libxslt-1.1.28-dev/bin:/nix/store/qs4ii32a46sn4y09f06pajcj6q2f2lbj-libxslt-1.1.28-bin/bin:/nix/store/j06s61q1jicacacnyzi4wp1wfmdkkarp-mythes-1.2.4/bin:/nix/store/z5mrfh0qqmlcrn2s71635pp3847fgl4l-gstreamer-1.8.2-dev/bin:/nix/store/k0pmsvylznrpjdbg9lpkw7z2q8189df6-gst-plugins-base-1.8.2/bin:/nix/store/8c05ki327bfzdpzvibkk1hghklcbx2fw-neon-0.30.1/bin:/nix/store/i9rjfg97y57bd36lp8yffdz727x4v4j0-nspr-4.12-dev/bin:/nix/store/vp7arfzv8x86skkck2sahbgzzb1ram1g-nss-3.23-dev/bin:/nix/store/lqk5s0v78gyprr1shj09xkwn3cqrs8l0-openldap-2.4.44/bin:/nix/store/1cl0ing62wlx5n0asbhnmc42afgfl9sk-ORBit2-2.14.19-dev/bin:/nix/store/vvma5jrd3wgw63sdirjxpplnac95nf8i-libIDL-0.8.14/bin:/nix/store/9z0iirjnqay56fpg6mjn1dq2fqqmssrb-ORBit2-2.14.19/bin:/nix/store/3vacnbzv0bp9si7rj5hnvmf1hmy26kxy-linux-pam-1.2.1/bin:/nix/store/ilv5vysh2hxbsxpjai017g43sg9k8c2f-perl-5.22.2/bin:/nix/store/7yrh96l3471vlbckcjqjdq8gi3l6qz98-pkg-config-0.29/bin:/nix/store/jrrqjx1hdcjdy8gcw2bb4zp95pnafc9j-openjpeg-2.1.0/bin:/nix/store/warfa9i7133nsr5nj4w8qkk484i61kkr-python3-3.5.2/bin:/nix/store/qii2a0gg8v2gipd23jin9vxf197rkn0z-sablotron-1.0.3/bin:/nix/store/ns9ljgmf9jsldq8cd62lj0167jvfnr8j-sane-backends-1.0.25/bin:/nix/store/fj4f4836cg542p3ryhpqdydhw0sp9r9b-tcsh-6.19.00/bin:/nix/store/nsr7gv1varvs6pkhjmmlxawjwq6pl0qv-unzip-6.0/bin:/nix/store/3ml94p4gic4zr0md08zfgqlqih0nz16l-vigra-1.10.0/bin:/nix/store/jjcmjxni6fnfygczgb6nw0w3jfsfjr4g-which-2.21/bin:/nix/store/77jhxclkzb3paqvhgxiv537bk669rkb2-zip-3.0/bin:/nix/store/7kxgnxx7hgy1ng3nsfq78dakhj6212xj-bluez-5.28/bin:/nix/store/mvgdvi0yiriczai2zdbvsm29ya8lgba5-libcmis-0.5.0/bin:/nix/store/blzzx4z0lrljj8br4ljl8msx7z7d9kxz-libwps-0.4.3/bin:/nix/store/issxa812rf85bynh8kh6q9csc0i2ycgy-libabw-0.1.1/bin:/nix/store/hq32df6ikc89lbndxn16k745hq5a78a6-harfbuzz-icu-1.2.7-dev/bin:/nix/store/7kla61mhl237j14a1fn48z9hz7f0mpqg-libe-book-0.1.2/bin:/nix/store/gs5vss14n2ghwcza49s0w9lr51qmmymd-libmwaw-0.3.7/bin:/nix/store/kaj1xyp72yv3a0l1vrsmh1xnl01s76zk-glew-1.13.0/bin:/nix/store/7g7rwc5ypxkr1mi59l7qq6j3d8wh0864-ncurses-6.0-dev/bin:/nix/store/c4j8zr7dlbs6wq5japmg3ms79w1yavgr-ncurses-6.0/bin:/nix/store/239l1g7bzlbh96h0wg3yv4c63473pdaj-CoinMP-1.7.6/bin:/nix/store/rm6x9vkx6r0spihd0r4x333irrm220kj-patchelf-0.9/bin:/nix/store/1pblq89nqzsz3prkmmzwv4lrq4qhfy3w-paxctl-0.9/bin:/nix/store/r8b456lwk0n1hinz92hwqsgb28c987b3-gcc-wrapper-5.4.0/bin:/nix/store/scfqn95555hsh9k1b0j1y1znzrkr2a5k-gcc-5.4.0/bin:/nix/store/84x0vldsif3v7i7c6i2ckfzp4jyg9bq9-binutils-2.26/bin:/nix/store/snxbisn3r0cb9k3xbdc73ady3msxb9wc-glibc-2.23-bin/bin:/nix/store/038z6c3idjsy5a4gkqb70s1vml7ydylw-coreutils-8.25/bin:/nix/store/038z6c3idjsy5a4gkqb70s1vml7ydylw-coreutils-8.25/bin:/nix/store/izh9gsfvi765hi89b0mwnfd58l4xgxd7-findutils-4.6.0/bin:/nix/store/0iz6w8v22iiypqjwlwyylvvpxmhwc8kv-diffutils-3.3/bin:/nix/store/r3m6xxf11rr65pyvfjslvj8lawpxjwi0-gnused-4.2.2/bin:/nix/store/8nm7vp5zw2lq3y2bvmq4yjnn2qgcm9rq-gnugrep-2.25/bin:/nix/store/0bby8n9ilkgd1fcwgcs5w9ymhx6vygls-gawk-4.1.3/bin:/nix/store/pgfnwxr4ky923p82qyjf8mf8mr7m3hvs-gnutar-1.29/bin:/nix/store/5nqba8ajfa3q91nvkyz7ddvvdyv8n7vq-gzip-1.8/bin:/nix/store/gcdzwrifzi2y9dg7mrg9ra4nm2phnkiz-bzip2-1.0.6.0.1-bin/bin:/nix/store/1di6pwjnapa7kj232dsidmz3klzrr0nd-gnumake-4.2.1/bin:/nix/store/b4xk5zs6w92xnzg851idlh2380401id1-bash-4.3-p42/bin:/nix/store/9y59gmsj24ppffpaxnr2bpclrv9w7kan-patch-2.7.5/bin:/nix/store/ca12bylrlmkdlzhi0k729xkrv68qx306-xz-5.2.2-bin/bin:/nix/store/jgn2whwk2a92y2wysb64kyrwfr5c9xkg-openjdk-8u102b04/lib/openjdk/bin)
You need gdb in your path to show backtraces

Error: a unit test failed, please do one of:

export DEBUGCPPUNIT=TRUE            # for exception catching
export CPPUNITTRACE="gdb --args"    # for interactive debugging on Linux
export VALGRIND=memcheck            # for memory checking

and retry using: make CppunitTest_sc_subsequent_filters_test

make[1]: *** [/tmp/nix-build-libreoffice-5.0.6.3.drv-0/libreoffice-5.0.6.3/solenv/gbuild/CppunitTest.mk:88: /tmp/nix-build-libreoffice-5.0.6.3.drv-0/libreoffice-5.0.6.3/workdir/CppunitTest/sc_subsequent_filters_test.test] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile:251: build] Error 2
builder for ‘/nix/store/cj5b5rz2g9rfv574yyidn04ypg4cv45l-libreoffice-5.0.6.3.drv’ failed with exit code 2
```

I didn't immediately find the 64-bit build failure on hydra, but here is the 32-bit failure (also test phase):
http://hydra.nixos.org/build/37518706/nixlog/3/raw

## Technical details

* System: x86_64-linux NixOS
* Nix version: 1.11.2
* Nixpkgs version: cde541236a076519ed83fd3b7ba1231779b84f93 (master as of today)

**Comment by bjornfor** [comment]
Fixed by 3223ef5df509cba22e5a9d77b33d4438ecd22a67. Thank you @7c6f434c!

**Comment by 7c6f434c** [comment]
>Fixed by 3223ef5df509cba22e5a9d77b33d4438ecd22a67. Thank you @7c6f434c!

I am not yet sure it was _the only_ problem…

**Comment by bjornfor** [comment]
The build passed on my system.

**Comment by 7c6f434c** [comment]
>The build passed on my system.

How interesting.

Both Still and Fresh fail on my buildbox (64-bit mode i7) — another
test.

**Comment by bjornfor** [comment]
Specifically:

```
$ nix-build -A libreoffice
/nix/store/xf3b0z0vr5hzkx9n8hqpa15pdhw7bl1l-libreoffice-5.0.6.3
(success)
```

On 64-bit NixOS, i5 CPU at nixpkgs 1b0e82d6ff86e4f9f072a16a71d443c2bea75633 (two commits ahead of your fix).

**Comment by 7c6f434c** [comment]
Hm, fontconfig may be relevant.

So my commit is before both the issue and before one of prerequisite commits. Tell me more about time going form past towards future.

**Comment by 7c6f434c** [comment]
Maybe fontconfig revert would be enough per se, though?

**Comment by bjornfor** [comment]
I'll try building libreoffice on 3223ef5df509cba22e5a9d77b33d4438ecd22a67 (your fix) and parent. (Neither includes the ```Revert "fontconfig: fix etc priority"``` commit.)

**Comment by bjornfor** [comment]
The build takes about 1.5 hours. I'm off to bed, will check it tomorrow.

**Comment by bjornfor** [comment]
@7c6f434c: Yes, you are correct: the fontconfig revert is sufficient. I tried to locally revert it (1e53d4a7776acbf61f42c094c103652c8068ad64) on top of the _parent_ of your fix (cde541236a076519ed83fd3b7ba1231779b84f93). And the build passes.

So, is your fix actually hiding real bugs then? The fontconfig change caused a "real bug" on my system, https://github.com/NixOS/nixpkgs/issues/16978 ("Font looks different + really bad kerning"), due to old state in /etc/fonts.

**Comment by 7c6f434c** [comment]
>@7c6f434c: Yes, you are correct: the fontconfig revert is sufficient. I tried to locally revert it (1e53d4a7776acbf61f42c094c103652c8068ad64) on top of the _parent_ of your fix (cde541236a076519ed83fd3b7ba1231779b84f93). And the build passes.
>
>So, is your fix actually hiding real bugs then? The fontconfig change caused a "real bug" on my system, https://github.com/NixOS/nixpkgs/issues/16978 ("Font looks different + really bad kerning"), due to old state in /etc/fonts.

Maybe. Feel free to revert.

Some of the very similar bugs happen even when everything observable is
fine, so I assumed it is a minor rendering glitch.

**Comment by bjornfor** [comment]
@7c6f434c: I'm not familiar with libreoffice, I don't want to decide on reverting or not. I'll leave that decision to you and others more qualified than me :-)
